configuration.go

```diff
@@ -47,13 +83,80 @@ func NewGlobalConfiguration() *GlobalConfiguration {
 	return globalConfiguration
 }
 
-// LoadFileConfig returns a GlobalConfiguration from reading the specified file (a toml file).
-func LoadFileConfig(file string) *GlobalConfiguration {
+// LoadConfiguration returns a GlobalConfiguration.
+func LoadConfiguration() *GlobalConfiguration {
 	configuration := NewGlobalConfiguration()
-	if _, err := toml.DecodeFile(file, configuration); err != nil {
+	viper.SetEnvPrefix("traefik")
+	viper.SetConfigType("toml")
+	viper.AutomaticEnv()
+	if len(viper.GetString("configFile")) > 0 {
+		viper.SetConfigFile(viper.GetString("configFile"))
+	} else {
+		viper.SetConfigName("traefik") // name of config file (without extension)
+	}
+	viper.AddConfigPath("/etc/traefik/")   // path to look for the config file in
+	viper.AddConfigPath("$HOME/.traefik/") // call multiple times to add many search paths
+	viper.AddConfigPath(".")               // optionally look for config in the working directory
+	err := viper.ReadInConfig()            // Find and read the config file
```

Comment:
Don't you think it's simpler here to handle error first?
Because using your notation:
```
if err := viper.ReadInConfig(); err != nil {
    // […]
}
else {
    fmtlog.Fatalf("Error reading file: %s", err)
}
```